src/cracker.c

```diff
@@ -1077,6 +1092,10 @@ static int crk_salt_loop(void)
 	return ext_abort;
 }
 
+/*
+ * Process an incomplete batch; This is used by mask mode before
+ * resetting the format with a changed internal mask.
+ */
 int crk_process_buffer(void)
```

Comment:
I think `crk_process_buffer()` is used by more than just mask mode, and existed before we got mask mode.

Reply:
No, I added it in c1d7a2f3 for mask mode for the very case documented. I'm not sure right now how the "last batch" of a run is processed, but it must be done by other means.